hoomd/AlchemyData.h

```diff
@@ -0,0 +1,173 @@
+                                     access_mode::read);
+        for (unsigned int i = 0; i < m_alchemical_derivatives.getNumElements(); i++)
+            netForce += h_forces.data[i];
+        // TODO: make clear the implementation choices being averaged quantities
+        netForce /= Scalar(m_alchemical_derivatives.getNumElements());
+        m_timestepNetForce.second = netForce;
+        }
+
+    void setNetForce(Scalar norm_value)
+        {
+        setNetForce();
+        m_timestepNetForce.second *= norm_value;
+        }
+
+    Scalar getNetForce(uint64_t timestep)
+        {
+        // TODO: remove this sanity check after we're done making sure timing works
+        assert(m_timestepNetForce.first == timestep);
+        return m_timestepNetForce.second;
+        }
+
+    Scalar getNetForce()
+        {
+        return m_timestepNetForce.second;
+        }
+
+    void setMass(Scalar new_mass)
+        {
+        mass.x = new_mass;
+        mass.y = Scalar(1.) / new_mass;
+        }
+
+    Scalar getMass()
+        {
+        return mass.x;
+        }
+
+    Scalar getValue()
+        {
+        return value;
+        }
+
+    pybind11::array_t<Scalar> getDAlphas()
+        {
+        ArrayHandle<Scalar> h_forces(m_alchemical_derivatives,
+                                     access_location::host,
+                                     access_mode::read);
+        return pybind11::array(m_alchemical_derivatives.getNumElements(), h_forces.data);
+        }
+
+    Scalar momentum = 0.; // the momentum of the particle
+    Scalar2 mass
+        = make_scalar2(1.0,
+                       1.0); // mass (x) and it's inverse (y) (don't have to recompute constantly)
+    Scalar mu = 0.;          //!< the alchemical potential of the particle
+    GlobalArray<Scalar> m_alchemical_derivatives; //!< Per particle alchemical forces
+    protected:
+    // the timestep the net force was computed and the netforce
+    std::pair<uint64_t, Scalar> m_timestepNetForce;
+    };
+
+// TODO: store a pointer to the force, to mirror the python, also add mass as an argument
+// TODO: add additional constructor that can work just off of python objects, maybe see
+// ComputeFreeVolume for the string conversion to type pair
+// parameter dict must be the same
+class AlchemicalPairParticle : public AlchemicalMDParticle
+    {
+    public:
+    AlchemicalPairParticle(std::shared_ptr<const ExecutionConfiguration> exec_conf,
+                           std::shared_ptr<ForceCompute> force_pair,
+                           int3 type_pair_param)
+        : AlchemicalMDParticle(exec_conf), m_type_pair_param(type_pair_param) {};
+    int3 m_type_pair_param;
+    };
+
+inline void export_AlchemicalMDParticle(pybind11::module& m)
+    {
+    pybind11::class_<AlchemicalMDParticle, std::shared_ptr<AlchemicalMDParticle>>(
+        m,
+        "AlchemicalMDParticle")
+        .def_property("mass", &AlchemicalMDParticle::getMass, &AlchemicalMDParticle::setMass)
```

Comment:
Also define properties (possibly read-only) for the `force`, `typepair`, and `name` python properties. The `typepair` and `name` getter and setter methods should convert between integers and strings